src/geneweaver/client/cli/alpha/api/genes.py

```diff
@@ -0,0 +1,43 @@
+"""Geneweaver API CLI for Gene endpoints."""
+
+import json
+from typing import List, Optional
+
+import typer
+from geneweaver.client.api import genes
+from geneweaver.client.auth import get_access_token
+
+cli = typer.Typer()
+
+HELP_MESSAGE = """
+Tools and utilities for interacting with the Geneweaver API.
+"""
+
+
+@cli.command()
+def map_ids(
+    ctx: typer.Context,
+    source_ids: List[str],
+    target_id_type: genes.GeneIdentifier,
+    source_id_type: Optional[genes.GeneIdentifier] = None,
+    target_species: Optional[genes.Species] = None,
+    source_species: Optional[genes.Species] = None,
+) -> dict:
+    """Map homologs between species."""
```

Comment:
args doc

Reply:
I leave the docstrings for the CLI endpoint as descriptive only because they are used in the auto-generated helpdocs. The `:param` lines would be redundant there.

e.g.

```
$ gweave alpha api genesets get --help
                                                                                            
 Usage: gweave alpha api genesets get [OPTIONS] GENESET_ID                                  
                                                                                            
 Get a Geneset by ID.                                                                       
                                                                                            
╭─ Arguments ──────────────────────────────────────────────────────────────────────────────╮
│ *    geneset_id      INTEGER  [default: None] [required]                                 │
╰──────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Options ────────────────────────────────────────────────────────────────────────────────╮
│ --gene-id-type        [Entrez|Ensemble Gene|Ensemble    [default: None]                  │
│                       Protein|Ensemble                                                   │
│                       Transcript|Unigene|Gene                                            │
│                       Symbol|Unannotated|MGI|HGNC|RGD|                                   │
│                       ZFIN|FlyBase|Wormbase|SGD|miRBas                                   │
│                       e|CGNC]                                                            │
│ --help                                                  Show this message and exit.      │
╰──────────────────────────────────────────────────────────────────────────────────────────╯
```

Reply:
```
$ gweave alpha api genes map-ids --help
                                                                                                
 Usage: gweave alpha api genes map-ids [OPTIONS] SOURCE_IDS...                                  
                                       TARGET_ID_TYPE:{Entrez|Ensemble                          
                                       Gene|Ensemble Protein|Ensemble                           
                                       Transcript|Unigene|Gene Symbol|Unannotat                 
                                       ed|MGI|HGNC|RGD|ZFIN|FlyBase|Wormbase|SG                 
                                       D|miRBase|CGNC}                                          
                                                                                                
 Map homologs between species.                                                                  
                                                                                                
╭─ Arguments ──────────────────────────────────────────────────────────────────────────────────╮
│ *    source_ids          SOURCE_IDS...                     [default: None] [required]        │
│ *    target_id_type      TARGET_ID_TYPE:{Entrez|Ensemble   [default: None] [required]        │
│                          Gene|Ensemble Protein|Ensemble                                      │
│                          Transcript|Unigene|Gene                                             │
│                          Symbol|Unannotated|MGI|HGNC|RGD|                                    │
│                          ZFIN|FlyBase|Wormbase|SGD|miRBas                                    │
│                          e|CGNC}                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────╯
╭─ Options ────────────────────────────────────────────────────────────────────────────────────╮
│ --source-id-type        [Entrez|Ensemble Gene|Ensemble    [default: None]                    │
│                         Protein|Ensemble                                                     │
│                         Transcript|Unigene|Gene                                              │
│                         Symbol|Unannotated|MGI|HGNC|RGD|                                     │
│                         ZFIN|FlyBase|Wormbase|SGD|miRBas                                     │
│                         e|CGNC]                                                              │
│ --target-species        [All|Mus musculus|Homo            [default: None]                    │
│                         Sapiens|Rattus Norvegicus|Danio                                      │
│                         Rerio|Drosophila                                                     │
│                         Melanogaster|Macaca                                                  │
│                         Mulatta|Caenorhabditis                                               │
│                         Elegans|Saccharomyces                                                │
│                         Cerevisiae|Gallus Gallus|Canis                                       │
│                         Familiaris]                                                          │
│ --source-species        [All|Mus musculus|Homo            [default: None]                    │
│                         Sapiens|Rattus Norvegicus|Danio                                      │
│                         Rerio|Drosophila                                                     │
│                         Melanogaster|Macaca                                                  │
│                         Mulatta|Caenorhabditis                                               │
│                         Elegans|Saccharomyces                                                │
│                         Cerevisiae|Gallus Gallus|Canis                                       │
│                         Familiaris]                                                          │
│ --help                                                    Show this message and exit.        │
╰──────────────────────────────────────────────────────────────────────────────────────────────╯

```